### ISSUE TITLE
Can we detect an image that does not have a true class?

### ISSUE THREAD

**Issue by off99555** [cgnorthcutt/cleanlab#32]
Suppose I'm classifying cats and dogs. But in the training data, there are sometimes images of a bird that are incorrectly labeled as cats or dogs. And this bird image cannot be labeled as dogs or cats. It's both wrong. The only course of action is to delete this image, you cannot fix the label.

Would `cleanlab` be able to detect bird images in a dataset of cats and dogs?

**Comment by cgnorthcutt** [comment]
Yes. Cleanlab does this automatically. If your classifier is appropriate for the data (reasonably high cross validation accuracy), it will include any image that is not labeled correctly in the error set, regardless of the underlying true class. 

This is easy to test. Add an image of the letter A into the MNIST dataset and see if it identified the letter as an error.

If your dataset contains lots of birds or if your model isn't very good (like if you use a simple naive Bayes on image data), then your model won't be able to train well enough on card and dogs to have good predicted probabilities, which will affect cleanlab's ability to find the errors.  But for the most part, if you use a reasonable model that has high accuracy (> 90 percent) on the data when it's clean and the data has no more than 40 percent error, cleanlab will find all the birds in your cat / dog set.

**Comment by off99555** [comment]
Thank you. This answered my question!
